Storage/SQL/BrowserDB.swift

```diff
@@ -430,12 +430,20 @@ extension BrowserDB: Queryable {
 
     func queryReturnsResults(sql: String, args: Args?=nil) -> Deferred<Maybe<Bool>> {
         return self.runQuery(sql, args: args, factory: { row in true })
-         >>== { deferMaybe($0[0] ?? false) }
+            >>== { if ($0.status == .Success) && ($0.count > 0) {
+                return deferMaybe($0[0] ?? false)
```

Comment:
This is a `Cursor<Bool>`, because the cursor factory just above is taking a row and returning `true`.

The specific implementation here is an `ArrayCursor`, which implements subscript:

```

    public override subscript(index: Int) -> T? {
        get {
            if (index >= data.count || index < 0 || status != .Success) {
                return nil
            }
            return data[index]
        }
    }
```

which reminds me why I didn't add the success check myself! So `cursor[0]` will return the first item — which our factory ensures is always `true` — if the query returned results and was successful. Otherwise it'll return `nil`, which `??` turns into `false`.

So you can skip the `Success` check altogether.